src/key_packages/mod.rs

```diff
@@ -455,6 +455,28 @@ impl KeyPackageBundle {
             leaf_secret,
         }
     }
+
+    #[cfg(any(feature = "expose-test-vectors", test))]
+    pub fn from_key_package_and_leaf_secret(
```

Comment:
This looks way to complex for something that's a test only function. Either we should actually use this function somewhere in the code or we shouldn't need it for thest.

Reply:
The problem is not the complexity of this code but that you are not testing any code that's actually being used. This should rather share code with `new_from_leaf_secret`, which is generating a `KeyPackageBundle` from a `leaf_secret` already.

Reply:
Ah, I see what you mean now. I think I tried that, but then I would need the `CredentialBundle` that was used to sign the `KeyPackage` to begin with and that's not part of the test vector. The way that the test vector is structured doesn't really match any of the code we use in practice. As a compromise I've restructured the code such that we now at least test the function that is used when refreshing an existing key package.